src/chrome/js/background.js

```diff
@@ -0,0 +1,10 @@
+chrome.tabs.onUpdated.addListener(
+  function(tabId, changeInfo, tab) {
+    if (changeInfo.url) {
+      chrome.tabs.sendMessage(tabId, {
+        message: 'url_changed',
+        url: changeInfo.url
+      });
+    }
+  }
+);
```

Comment:
The hardest part of this was figuring out how to get data as soon as a battle starts instead of just when the plugin loading script is initialized.

My solution is to use the URL update as a trigger to check the tier (which is in the URL). When the tier changes, the extension fires off a custom event that our JS code listens for and then fetches the latest data.